### PR TITLE
Fix error in zsh si script

### LINKED ISSUE #188875: Prepending fails on zsh with only shell integration enabled
Testing #182970

```typescript
envVarCollection.prepend('PS1', '(.venv) ', {
    applyAtProcessCreation: false,
    applyAtShellIntegration: true,
});
```
seems to fail on zsh shell:

image

### PR DESCRIPTION
**Pull request by Tyriar**
Fixes #188875

Screenshot 2023-08-30 at 10 05 16 AM

### CODE REVIEW

**Review by karrtikr — COMMENTED** [review]
Nice! Although there does not seem be a space between the prepended character and rest of the string.

**Comment by Tyriar** [comment]
@karrtikr that's by design as my test extension prepended `"(.venv)"`, not `"(.venv) "`

**Review by karrtikr — APPROVED** [review]
Ah okay

File: src/vs/workbench/contrib/terminal/browser/media/shellIntegration-rc.zsh [hunk 1]
```diff
@@ -51,15 +51,15 @@ if [ -n "${VSCODE_ENV_PREPEND:-}" ]; then
...
+		export $VARNAME="$(echo -e ${ITEM#*=})${(P)VARNAME}"
 	done
 	unset VSCODE_ENV_PREPEND
 fi
 if [ -n "${VSCODE_ENV_APPEND:-}" ]; then
 	IFS=':' read -rA ADDR <<< "$VSCODE_ENV_APPEND"
 	for ITEM in "${ADDR[@]}"; do
 		VARNAME="$(echo ${ITEM%%=*})"
-		export $VARNAME="${(P)VARNAME}$(echo -e {ITEM#*=})"
+		export $VARNAME="${(P)VARNAME}$(echo -e ${ITEM#*=})"
```
Comment (RadaHugo7) [inline src/vs/workbench/contrib/terminal/browser/media/shellIntegration-rc.zsh:62]: []()

File: src/vs/workbench/contrib/terminal/browser/media/shellIntegration-rc.zsh [hunk 2]
```diff
@@ -51,15 +51,15 @@ if [ -n "${VSCODE_ENV_PREPEND:-}" ]; then
 	IFS=':' read -rA ADDR <<< "$VSCODE_ENV_PREPEND"
 	for ITEM in "${ADDR[@]}"; do
 		VARNAME="$(echo ${ITEM%%=*})"
-		export $VARNAME="$(echo -e {ITEM#*=})${(P)VARNAME}"
+		export $VARNAME="$(echo -e ${ITEM#*=})${(P)VARNAME}"
 	done
```
Comment (RadaHugo7) [inline src/vs/workbench/contrib/terminal/browser/media/shellIntegration-rc.zsh:55]:
```suggestion
	done
```

**Pull request merged**